lib/graph.js

```diff
@@ -11,8 +11,8 @@ class Graph {
         return this.built_graph.inputs;
     }
 
-    add_node(type, name) {
-        var index = this.get_nodes().length;
+    add_node(type, name, index) {
```

Comment:
I’d slightly prefer to have a separate `insert_node(type, name, index)` method and to reimplement `add_node(type, name)` in terms of it. The idea is that appending and inserting at a specific position are two different things so they should be serviced by two different methods.